pkg/proxy/iptables/proxier.go

```diff
@@ -296,6 +318,15 @@ func NewProxier(ipFamily v1.IPFamily,
 		nodePortAddresses:        nodePortAddresses,
 		networkInterfacer:        proxyutil.RealNetwork{},
 		conntrackTCPLiberal:      conntrackTCPLiberal,
+		nfAcctTrackers: map[string]*nfAcctTracker{
+			metrics.IptablesCTStateInvalidDroppedPackets.Name: {
+				name:    base32SHA256Sum(metrics.IptablesCTStateInvalidDroppedPackets.Name, string(ipFamily))[:nfacct.MaxLength],
+				ensured: false,
+				updateMetric: func(packets, _ uint64) {
+					metrics.IptablesCTStateInvalidDroppedPackets.WithLabelValues(string(ipFamily)).Set(float64(packets))
+				},
+			},
+		},
```

Comment:
this is much bigger than any of the other fields in the `&Proxier{}` constructor here... I'd assign this separately after the main `proxier :=`

(You could even do

```
for _, metric := range []kmetrics.Metric{metrics.IptablesCTStateInvalidDroppedPackets} {
        proxier.nfAcctTrackers[metric.Name] = &nfAcctTracker{
                name:    base32SHA256Sum(metric.Name, string(ipFamily))[:nfacct.MaxLength],
                ensured: false,
                updateMetric: func(packets, _ uint64) {
                        metric.WithLabelValues(string(ipFamily)).Set(float64(packets))
                },
        }
}
```

although I guess that assumes that all of the metrics will have the same form, in which case you don't need `updateMetric`...

Reply:
the logic behind `updateMetric` function was that in future maybe some metric wanted to track bytes instead of packets...